app/models/state_file_base_intake.rb

```diff
@@ -118,6 +119,25 @@ def synchronize_df_1099_rs_to_database
     end
   end
 
+  def synchronize_df_w2s_to_database
+    direct_file_data.w2s.each_with_index do |direct_file_w2, i|
+      state_file_w2 = state_file_w2s[i] || state_file_w2s.build
```

Comment:
are we certain that `state_file_w2s` will be indexed the same way as the order of w2s in xml? Do we need to do any kind of "matching" like we do on dependents? Maybe on employerStateIdNum and EmployeeNm?

Reply:
oh good catch! we actually have a column on state_file_w2 called w2_index that we can use here that corresponds to the index of a w2 in the direct file xml